### DISCUSSION TITLE
Configing `fortls` failed

### DISCUSSION THREAD

**Discussion by himcraft** [fortran-lang/fortls#148] [Q&A] [answered]
Hi Mr. Nikiteas,

I was trying to add the support of suffix `.cuf` to `fortls` following the documents on [the official website](https://gnikit.github.io/fortls/options.html#configuration-using-the-command-line), but after I typed `fortls --incl_suffixes ".cuf"` and hit Enter, the prompt moved to the next line, but there was no more response on the screen. Then I hit Enter again, and it throws out an error as follows:
```
Content-Length: 125
Content-Type: application/vscode-jsonrpc; charset=utf8

{"jsonrpc":"2.0","method":"window/showMessage","params":{"type":1,"message":"Unexpected error: Line endings must be \\r\\n"}}Unexpected error: Line endings must be \r\n
Traceback (most recent call last):
  File "/home/me/.local/lib/python3.8/site-packages/fortls/langserver.py", line 114, in run
    request = self.conn.read_message()
  File "/home/me/.local/lib/python3.8/site-packages/fortls/jsonrpc.py", line 102, in read_message
    return self._receive()
  File "/home/me/.local/lib/python3.8/site-packages/fortls/jsonrpc.py", line 87, in _receive
    length = self._read_header_content_length(line)
  File "/home/me/.local/lib/python3.8/site-packages/fortls/jsonrpc.py", line 74, in _read_header_content_length
    raise JSONRPC2ProtocolError("Line endings must be \\r\\n")
fortls.jsonrpc.JSONRPC2ProtocolError: Line endings must be \r\n
```

I searched this error and found it is related to the difference between Unix and Dos, but I tried this on Ubuntu 20.04 in bash environment. 

And I cannot config correctly on Windows platform as well. On Windows, it just does not print anything no matter how many Enters I hitted.

Any suggestions please? Thank you!

**Comment by gnikit** [comment, marked as answer]
The way that you use `fortls` is through your editor. You do not spawn `fortls` yourself from a terminal. The first time you press enter it basically starts the Fortran Language Server but no code editor is listening so nothing happens. When you press Enter the second time you basically interfere with the already spawned server and it causes it to malfunction.

In your code editor e.g. VS Code, you will need to add `.cuf` as an extension to the server

**Reply by himcraft** [reply]
I see, as the `fortls` does not fully support CUDA Fortran yet, the kernel functions are not properly recognized, while I thought this was because I have not included the suffix `cuf` to `fortls`'s configuration. Thank you!
